state_machine/state_engine.py

```
@@ -121,45 +122,50 @@ def set_phase(self, phase: MissionPhase) -> Dict[str, Any]:
+                    target_phase_str = phase
+                else:
+                     raise ValidationError(f"Invalid phase type: {type(phase)}")
+
+                # Use central validator for phase name
+                target_phase_str = MissionPhaseValidator.validate_phase(target_phase_str)
+                
+                # Convert back to Enum if needed for internal consistency
+                try:
+                    target_phase_enum = MissionPhase(target_phase_str)
+                except ValueError:
+                     raise ValidationError(f"Phase {target_phase_str} not found in MissionPhase enum")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)